lua/nvim-treesitter/shell_command_selectors.lua

```diff
@@ -96,8 +96,14 @@ function M.select_compiler_args(repo, compiler)
       repo.files,
       "-shared",
       "-Os",
-      "-lstdc++",
     }
+    if
+      #vim.tbl_filter(function(file)
+        return file:match "%.cc$" or file:match "%.cpp$" or file:match "%.cxx$"
```

Comment:
This should be more performant but it probably doesn't matter much.
```suggestion
        local ext = vim.fn.fnamemodify(file, ":e")
        return ext == "cc" or ext == "cpp" or ext == "cxx"
```

Reply:
I wouldn't do speculations on performance when we use a vim-script function that requires conversion. But it reads clearer I guess (apart from magic happening with `:e`)

Reply:
> I wouldn't do speculations on performance when we use a vim-script function that requires conversion.

We could use `vim.split` instead. The point was to avoid running 3 pattern matches.